### PR TITLE
Additional test for queues

### PR DESCRIPTION
**Pull request by sopel39**
Test that verifies complex queues configuration
scenario was added.

### CODE REVIEW

**Comment by sopel39** [comment]
@cberner additional tests

File: presto-main/src/test/java/com/facebook/presto/execution/TestSqlQueryQueueManager.java [hunk 1]
```diff
@@ -1,55 +0,0 @@
...
-
-import static org.testng.Assert.assertTrue;
-import static org.testng.Assert.fail;
-
-public class TestSqlQueryQueueManager
-{
-    @Test
-    public void testJsonParsing()
-    {
-        parse("queue_config.json");
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/execution/TestSqlQueryQueueManager.java]: I think the parsing tests should stay here in `presto-main` as they are unit tests, and it is better to have them in the same module to detect bugs earlier.

File: presto-tests/src/test/java/com/facebook/presto/execution/TestSqlQueryQueueManager.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+    @Test
+    public void testJsonParsing()
+    {
+        parse("queue_config.json");
+        assertFails("queue_config_bad_cycle.json", "Queues must not contain a cycle. The shortest cycle found is \\[q(.), q., q., q., q\\1\\]");
+        assertFails("queue_config_bad_selfcycle.json", "Queues must not contain a cycle. The shortest cycle found is \\[q1, q1\\]");
+    }
+
+    @Test(timeOut = 240_000)
```
Comment (electrum) [inline presto-tests/src/test/java/com/facebook/presto/execution/TestSqlQueryQueueManager.java]: How long does this test actually take?

File: presto-tests/src/test/resources/queue_config_dashboard.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      ]
+    },
+    {
+      "queues": [
+        "user.${USER}",
+        "global"
+      ]
+    }
+  ]
+}
```
Comment (electrum) [inline presto-tests/src/test/resources/queue_config_dashboard.json]: Missing trailing newline

File: presto-main/src/test/java/com/facebook/presto/execution/TestSqlQueryQueueManager.java (hunk 1 above)
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/execution/TestSqlQueryQueueManager.java]: I have restored unit test

File: presto-tests/src/test/java/com/facebook/presto/execution/TestSqlQueryQueueManager.java (hunk 2 above)
Comment (sopel39) [inline presto-tests/src/test/java/com/facebook/presto/execution/TestSqlQueryQueueManager.java]: 20 seconds on my machine.

File: presto-tests/src/test/resources/queue_config_dashboard.json (hunk 3 above)
Comment (sopel39) [inline presto-tests/src/test/resources/queue_config_dashboard.json]: added

**Comment by sopel39** [comment]
I've rebased and applied comments

**Comment by cberner** [comment]
It looks like the test is failing in Travis. Can you take a look to see if the test timeout is too short?

**Comment by sopel39** [comment]
@cberner It seems that Travis has log limit of 4MB per build. I have updated the test so that it produces much less output (comparable to TestMemoryManager), but there is still some unnecessary debug messages. Do you know to to disable airlift logger for a particular class and test?

**Comment by cberner** [comment]
Hmm, I'm not sure either. @electrum or @martint do you know if that's possible?

**Comment by sopel39** [comment]
Anyway, the tests pass now

**Comment by cberner** [comment]
Made a couple methods static and merged. Thanks!

**Pull request closed without merging**